pkg/client/leaderelection/resourcelock/interface.go

```diff
@@ -69,3 +74,29 @@ type Interface interface {
 	// into a string
 	Describe() string
 }
+
+// Manufacture will create a lock of a given type according to the input parameters
+func Manufacture(lockType string, ns string, name string, client *cs.Clientset, rlc ResourceLockConfig) (Interface, error) {
```

Comment:
Why not just call this New? What is the significance of Manufacture in this context?